spec.md

```diff
@@ -721,14 +721,10 @@ A client querying the [referrers API](#listing-referrers) and receiving a `404 N
 
 ##### Referrers Tag Schema
 
-```text
-<alg>-<ref>
-```
-
-- `<alg>`: the digest algorithm (e.g. `sha256` or `sha512`)
-- `<ref>`: the digest from the `subject` field (limit of 64 characters)
+The Referrers Tag associated with a [Content Digest](https://github.com/opencontainers/image-spec/blob/v1.0.1/descriptor.md#digests) <sup>[apdx-3](#appendix)</sup> MUST match the digest truncated to 128 characters with any characters not allowed by [`<reference>` tags](#pulling-manifests) replaced with `-`.
```

Comment:
Changing to a 128 character limit here means client tooling needs a rewrite. And any content pushed with the previous 64 character hash limit would no longer be seen. It's unlikely that there's content out there today, but I'm not seeing a big enough value add to having incompatible clients between spec releases. I'd go back to the 64 character hash limit, clarifying that limit is achieved via truncation.

Reply:
Do existing clients actually implement `encoded` section truncation at all?  Because `sha256` was already 64 characters, so they might have assumed it was already short enough?

There's also no [image-spec-side](https://github.com/opencontainers/image-spec/blob/v1.0.1/descriptor.md#digests) limit on the length of the `algorithm` section, so it's possible that 128 char truncation of the overall digest (to fit into the tag limit) means the `encoded` part would have to be truncated at fewer than 64 characters, possibly zero characters in some extremely-unlikely-but-formally-compliant 128-character `algorithm`.

But sure, I can probably figure out a way to say that `encoded` needs 64-char truncation, and then the truncated-encoded digest needs 128-char truncation...

Reply:
I'm not at my desk right now, so off the top of my head is my own project: 
https://github.com/regclient/regclient/blob/main/types%2Freferrer%2Freferrer.go#L157

Not sure about other projects like oras and go-containerregistry.

Reply:
Something else to consider is whether it would be easier to limit the digest algorithm (length and character set) given how few implementations there are of that. Because a 120 character long digest algorithm would cause lots of trouble.

Reply:
> Something else to consider is whether it would be easier to limit the digest algorithm (length and character set)...

Seems reasonable to me, but also seems like an image-spec issue.  Because bolting on new constraints at the distribution-spec would leave space for OCI-compatible images that were blocked from OCI-compatible distribution.

Reply:
Hmm, although sadly for the image-spec folks [their current][1] [EBNF][2] doesn't seem to support "between *n* and *m* characters" constructions, and they might need to pivot to [ABNF][3] or another syntax-specification format that does support more formal repetition specification.

[1]: https://github.com/opencontainers/image-spec/blame/v1.0.1/descriptor.md#L69
[2]: https://github.com/opencontainers/image-spec/blame/v1.0.1/considerations.md#L28-L103
[3]: https://www.rfc-editor.org/rfc/rfc5234.html#section-3.7

Reply:
pulling [this feedback][1] back into this thread, so "what actually should be the tag for a given digest?" discussion is collected in one place, without getting interleaved with other ideas.  I've pushed fffa895 -> e839150 to drop the overall 128 truncation and replace it with the 32-and-64 truncation.

[1]: https://github.com/opencontainers/distribution-spec/pull/563#pullrequestreview-2599659231